src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -718,6 +728,17 @@ void GoogleCalendarSyncAdaptor::eventsFinishedHandler()
         // we've finished loading all pages of event information
         // we now need to process the loaded information to determine
         // which events need to be added/updated/removed locally.
+        QDateTime since = needCleanSync ? QDateTime()
+                                        : lastSyncTimestamp(QLatin1String("google"),
+                                                            SocialNetworkSyncAdaptor::dataTypeName(SocialNetworkSyncAdaptor::Calendars),
+                                                            accountId).addSecs(2); // add 2 secs to avoid fs sync time issues.
+
+        if (!updated.isEmpty()) {
+            m_idDb.setLastUpdateTime(calendarId, accountId, updated);
+            TRACE(SOCIALD_DEBUG,
+                  QString(QLatin1String("Setting updated timestamp for Google account: %1. Calendar Id: %2.  Timestamp: %3."))
+                     .arg(accountId).arg(calendarId).arg(updated));
+        }
         updateLocalCalendarNotebookEvents(accountId, accessToken, calendarId, since);
```

Comment:
We should use 'since' because the value is used for building a list of events which have been locally modified since the last buteo sync and event modification timestamp in mkal db is in device time.